osu.Game/Utils/GeometryUtils.cs

```diff
@@ -218,5 +219,137 @@ private static IEnumerable<Vector2> enumerateStartAndEndPositions(IEnumerable<IH
 
                 return new[] { h.Position };
             });
+
+        #region welzl_helpers
+
+        // Function to check whether a point lies inside or on the boundaries of the circle
+        private static bool isInside((Vector2, float) c, Vector2 p)
+        {
+            return Precision.AlmostBigger(c.Item2, Vector2.Distance(c.Item1, p));
+        }
+
+        // Function to return a unique circle that intersects three points
+        private static (Vector2, float) circleFrom(Vector2 a, Vector2 b, Vector2 c)
+        {
+            if (Precision.AlmostEquals(0, (b.Y - a.Y) * (c.X - a.X) - (b.X - a.X) * (c.Y - a.Y)))
+                return circleFrom(a, b);
+
+            // See: https://en.wikipedia.org/wiki/Circumscribed_circle#Cartesian_coordinates_2
```

Comment:
Link doesn't seem related (anymore)? There is no "Cartesian coordinates 2" section in the linked article.